staging/src/k8s.io/cli-runtime/pkg/resource/helper.go

```diff
@@ -142,6 +144,21 @@ func (m *Helper) Patch(namespace, name string, pt types.PatchType, data []byte,
 	if options == nil {
 		options = &metav1.PatchOptions{}
 	}
+
+	obj, err := runtime.Decode(unstructured.UnstructuredJSONScheme, data)
```

Comment:
That patch can be in many formats, I doubt you can decode them with this function.

Reply:
So how would you decode the patch? Is there a reference that has all the acceptable formats of `data`?

Reply:
I don't think you really can